agent/taskresource/envFiles/envfile.go

```diff
@@ -452,3 +458,50 @@ func (envfile *EnvironmentFileResource) UnmarshalJSON(b []byte) error {
 
 	return nil
 }
+
+// ReadEnvVarsFromEnvFiles reads the environment files that have been downloaded
+// and puts them into a list of maps
+func (envfile *EnvironmentFileResource) ReadEnvVarsFromEnvfiles() ([]interface{}, error) {
+	var envVarsPerEnvfile []interface{}
+
+	for _, envfilePath := range envfile.environmentFilesLocation {
+		envVars, err := envfile.readEnvVarsFromFile(envfilePath)
+		if err != nil {
+			return nil, err
+		}
+		envVarsPerEnvfile = append(envVarsPerEnvfile, envVars)
+	}
+
+	return envVarsPerEnvfile, nil
+}
+
+func (envfile *EnvironmentFileResource) readEnvVarsFromFile(envfilePath string) (map[string]string, error) {
+	file, err := envfile.os.Open(envfilePath)
+	if err != nil {
+		seelog.Errorf("Unable to open environment file at %s to read the variables", envfilePath)
+		return nil, err
+	}
+	defer file.Close()
+
+	scanner := bufio.NewScanner(file)
+	envVars := make(map[string]string)
+	for scanner.Scan() {
+		line := scanner.Text()
+		// if line starts with a #, ignore
+		if strings.HasPrefix(line, commentIndicator) {
+			continue
+		}
+		// only read the line that has "="
+		if strings.Contains(line, envVariableDelimiter) {
+			variables := strings.Split(line, "=")
```

Comment:
this will read a line that has "=" anywhere, e.g. strings.Split("=xyz ", "=") returns ["", "xyz"]. Im inclined to use regex or any other validation that makes sure either side of = is at least length of one.

Reply:
we decided not to add any agent specific validations, so at least we should understand what happens when invalid line is passed to docker.

Reply:
good point - and is the existing logic to ignore any lines that is a comment or doesn't match the regex ok?

Reply:
im ok with that

Reply:
instead of regex i added a super simple "if both sides length > 0" lmk if you have any concerns, i'm open to adding it as regex